docs/cli/plugin_implementation_guide.md

```diff
@@ -39,12 +41,55 @@ filesystem repository in the creation and testing of their feature.
 
 #### Adding Admin Repository
 
+NOTE: This is not applicable if context-aware plugin discovery is enabled within Tanzu CLI.
+
 The admin repository contains the Builder plugin - a plugin which helps scaffold and compile plugins.
 
 To add the admin repository use `tanzu plugin repo add -n admin -b tanzu-cli-admin-plugins -p artifacts-admin`
 
 To add the builder plugin use `tanzu plugin install builder`
 
+#### Plugin Discovery Source
+
+Discovery is the interface to fetch the list of available plugins, their supported versions and how to download them either standalone or scoped to a context(server). E.g., the CLIPlugin API in a management cluster, OCI based plugin discovery for standalone plugins, a similar REST API etc. Having a separate interface for discovery helps to decouple discovery (which is usually tied to a server or user identity) from distribution (which can be shared).
```

Comment:
Since this is the plugin implementation guide, I think it might also help to say a bit about how to decide which discovery approach is appropriate for a given plugin. Something to help authors understand whether they should be leveraging the CLIPlugin API (putting an instance of that resource into some package) or creating a new standalone source.

Reply:
Added more details about the type of plugins and a use case example for context scoped plugins.